devito/operator.py

```diff
@@ -133,7 +134,9 @@ def expr_cse(expr):
     new_stencils = []
 
     for stencil in stencils:
-        new_stencils.append(stencil)
+        new_stencils.append(Eq(stencil.lhs,
+                               collect(stencil.rhs,
+                                       [temp.lhs for temp in to_keep])))
```

Comment:
This is the factorization. Mainly once the `temp` are found by cse, I force the stencils to be factorized by it as it chould be some sort of `common factors`. It seems to work quite well for acoustic wich end up doing a radial factorization (distance to (x,y,z)).